components/admin_console/workspace-optimization/dashboard.tsx

```diff
@@ -234,7 +234,8 @@ const WorkspaceOptimizationDashboard = (props: Props) => {
         current: 0,
     };
 
-    const accData: AccordionItemType[] = Object.entries(data).map(([accordionKey, accordionData]) => {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const accData: AccordionItemType[] = Object.entries(data).filter(([_, y]) => !y.hide).map(([accordionKey, accordionData]) => {
```

Comment:
to prevent disabling the eslint rule we should probably filter out hidden entries before or inside the `Object.entries` call:

```suggestion
    const accData: AccordionItemType[] = Object.entries(data.filter((x) => !x.hide)).map(([accordionKey, accordionData]) => {
```

Reply:
the suggested code won't work because of the data structure. The other suggestion about doing it before getting here  IMO adds more code complexity and will still require the key/value data structure, so basically the same. I rather keep it this way 4/5 just disabling the rule for this line, I think the code is simpler and easy to follow the logic.

Reply:
Yes, you are right ... my suggestion is a bit less readable.
I would probably filter out the hidden entries before and store them in a variable

```suggestion
    const filteredData = data.filter((item) => !item.hide);
    const accData: AccordionItemType[] = Object.entries(filteredData).map(([accordionKey, accordionData]) => {
```

Reply:
this won't work either without having to force cast several values adding more boilerplate than just disabling the rule, and probably to make it work and make happy the ts-es-linters will have to disable the eslint rule too, since the DataType is a Dictionary type. I still go with the initial implementation and just disabling the rule for that particular line.

Reply:
The only other solution I see for avoiding to disable the line for the rule is to reduce the function, but I don't see the value of making a considerable size refactoring and with potential regressions for just avoiding the disable of one linter rule for typescript which in this particular case is just syntactic sugar and won't cause any kind of problem.

Reply:
yes, after checking it out and testing for myself it does seem like the most reasonable approach.